File: oracles/Java Oracles/04 - Identifier/Identifier.java
public class Identifier {

	public static boolean valid_s(char c) {
		if (((c >= 'A') &&
			(c <= 'Z')) ||
			((c >= 'a') &&
			(c <= 'z')))
			return true;
		else
			return false;
	}

	public static boolean valid_f(char c) {
		if (((c >= 'A') &&
			(c <= 'Z')) ||
			((c >= 'a') &&
			(c <= 'z')) ||
			((c >= '0') &&
			(c <= '9')))
			return true;
		else
			return false;
	}

	public static boolean identifier(String str) {
		if(str.length() >= 1 && str.length() <= 6 && valid_s(str.charAt(0))) {
			for(int i = 1; i < str.length(); i++)
				if(!valid_f(str.charAt(i)))
					return false;
			return true;
		}
		return false;
	}

}

File: resources/oracles/C/Identifier/Identifier.c
#include "Identifier.h"

unsigned char valid_s(char c) {
	if (((c >= 'A') &&
		(c <= 'Z')) ||
		((c >= 'a') &&
		(c <= 'z')))
		return;
	else
		return 0;
}

unsigned char valid_f(char c) {
	if (((c >= 'A') &&
		(c <= 'Z')) ||
		((c >= 'a') &&
		(c <= 'z')) ||
		((c >= '0') &&
		(c <= '9')))
		return 1;
	else
		return 0;
}

unsigned char identifier(char *str) {
	unsigned char n, i, valid_id;
	n = strlen(str);
	valid_id = valid_s(str[0]);
	if (n > 1)
		for(i = 1; i < n; i++)
			if(!valid_f(str[i]))
				valid_id = 0;
	if ((n >= 1) && (n <= 6) && valid_id)
		return 1;
	else
		return 0;
}
